src/Symfony/Component/AssetMapper/AssetMapperCompiler.php

```diff
@@ -25,20 +26,25 @@ class AssetMapperCompiler
     /**
      * @param iterable<AssetCompilerInterface> $assetCompilers
      */
-    public function __construct(private iterable $assetCompilers)
+    public function __construct(private iterable $assetCompilers, private ContainerInterface $container)
```

Comment:
This is the ugliest part of this PR, and it's actually a small fix so that decorating the AssetMapper will work properly.

* It's super convenient for compilers to have access to the AssetMapper
* But, we get a circular reference: AssetMapper -> MappedAssetFactory -> AssetMapperCompiler -> AssetMapper

Is there a more elegant way to make *just* this one dependency lazy?

Reply:
> But this would make the component much harder to use standalone

Thank you - that makes me want to keep this as it is :)

Reply:
what about using a closure here, that'd return the mapper? that's a service_closure inthe DI world

Reply:
I don't think we can keep this as is TBH.

Reply:
I'll fix this up shortly!